sql/codegen.go

```diff
@@ -309,37 +309,36 @@ print("Training set accuracy: {accuracy:0.5f}".format(**eval_result))
 print("Done training")
 {{- else}}
 
-def pred_input_fn(batch_size):
-	feature_types = dict()
-	feature_shapes = dict()
-	for name in feature_column_names:
-		if column_name_to_type[name] == "categorical_column_with_identity":
-			feature_types[name] = tf.int64
-			feature_shapes[name] = tf.TensorShape([None])
-		else:
-			feature_types[name] = tf.float32
-			feature_shapes[name] = tf.TensorShape([])
-
-	gen = db_generator_predict(driver, conn, """{{.StandardSelect}}""",
-		feature_column_names, column_name_to_type)
-	dataset = tf.data.Dataset.from_generator(gen, feature_types, feature_shapes)
-	dataset = dataset.batch(batch_size)
-	return dataset
+def eval_input_fn(features, batch_size):
+	dataset = tf.data.Dataset.from_tensor_slices(dict(features))
+	return dataset.batch(batch_size)
 
+field_names, columns = execute(driver, conn, """{{.StandardSelect}}""")
 X = {}
```

Comment:
I think this is just a hotfix right, maybe should refine the `insert` function to insert predict result with original feature data.

Reply:
Sure. I just recovered it by the former code here.